### DISCUSSION TITLE
VPN plugin

### DISCUSSION THREAD

**Discussion by antoinekh** [nautobot/nautobot#4531] [General]
Hello,

Could be very nice to have a plug-in for site to site VPN tunnels.

We managed hundreds IPsec tunnels with customer and it will be perfect if all this information could also be in Nautobot

This is a list of fields that I'd like to have

- Name
- status (Active, Offline, Backup, Planned, Deprecated)
- type (route based VTI, policy based)
- nat traversal enabled (true, false)
- interface of the device (linked to device interface)
- local peer ip (one or more, linked to ip in IPAM)
- remote peer ip (one or more, linked to ip in IPAM)
- list of local subnets (linked to Subnets in IPAM)
- list of remote subnets (linked to Subnets in IPAM)
- protocols used (IKEv1/IKEv2 and IPsec)
- date created
- contact field
- description field

…

**Comment by chadell** [comment]
Hi @antoinekh , we (@Kircheneer ) are currently working on a model extension for that, as a Nautobot App. 
Hopefully, we will make it available very soon.

**Reply by jo-bzh** [reply]
Hello @chadell and @Kircheneer,
Just for us in order to have a visibility of what is inside your development, could you give us a way to see the functional and technical aspect of this app ?
Due to short term expectation on our side, we are going to develop a custom plugin to answer our needs.
This is why it could be helpful to be closed as possible to your app in order to migrate when the stable version will be released.
Thanks by advance.

**Reply by Kircheneer** [reply]
Hi! I meant to post [this](https://github.com/orgs/nautobot/discussions/4531#discussioncomment-7324040) as a comment here.

**Comment by Kircheneer** [comment]
We are planning on the following modelling.

# `Tunnel`

Has name, description and status as well as a `TunnelProfile` and 0-2 `TunnelEndpoint` instances, one for each end.

# `TunnelEndpoint`

Each tunnel has 0-2 of these - they map the tunnel into the rest of the models:

- Overlay interface (e.g. `Tunnel` interface on Cisco gear)
- Underlay interface (i.e. the interface that holds the IP address for the tunnel)
- Source IP address (the destination can be found on the other endpoint)
- Advertised prefixes

# `TunnelProfile`

Has name, description and a JSON blog for variable settings (like NAT traversal).

**Reply by jo-bzh** [reply]
Hello,
Thanks for your proposal,
The expected model is really the model proposed on the v3.7 of Netbox [9816-tunnels](https://github.com/netbox-community/netbox/pull/14276/commits).

**Comment by cardoe** [comment]
Related https://github.com/nautobot/nautobot/issues/915

**Comment by chadell** [comment]
VPN tunnel models will appear soon, likely in the next minor/major release: https://github.com/nautobot/nautobot/tree/feature/vpn-models
Other related models will come later.

**Reply by JHarrison712** [reply]
Any update on this one?

**Reply by chadell** [reply]
it's coming in the Nautobot 3.0 release: https://github.com/nautobot/nautobot/pull/7355

**Reply by christianbur** [reply]
When will version 3.0 be released?
Version 3.0 is already listed here: https://docs.nautobot.com/projects/core/en/next/release-notes/version-3.0/

I found this too:
https://github.com/iss-world/nautobot-app-vpn

**Reply by JHarrison712** [reply]
> it's coming in the Nautobot 3.0 release: #7355

Do you know when v3.0 will be released or hits beta?
